amgut/lib/data_access/test/test_ag_data_access.py

```diff
@@ -14,16 +19,114 @@ def test_authenticateWebAppUser(self):
         raise NotImplementedError()
 
     def test_addAGLogin(self):
-        raise NotImplementedError()
+        # test new user
+        exists = 'EXISTS'
+        while exists is not None:
+            email = ''.join([choice(ascii_letters)
+                             for i in range(randint(5, 10))])
+            domain = ''.join([choice(ascii_letters)
+                             for i in range(randint(5, 10))])
+            new_email = '@'.join([email, domain]) + '.com'
+            exists = self.ag_data.check_login_exists(new_email)
+
+        # make sure the ag_login_id is a UUID4 string
+        ag_login_id = self.ag_data.addAGLogin(
+            new_email, 'TESTDUDE', '123 fake test street', 'testcity',
+            'teststate', '1L2 2G3', 'United Kingdom')
+        as_uuid = UUID(ag_login_id)
+        self.assertTrue(as_uuid.version, 4)
+
+        # test existing user
+        ag_login_id = self.ag_data.addAGLogin(
+            'TEST@EMAIL.com', 'TESTOTHER', '123 fake test street', 'testcity',
+            'teststate', '1L2 2G3', 'United Kingdom')
+
+        obs = self.ag_data.addAGLogin(
+            'test@EMAIL.com', 'TESTDUDE', '123 fake test street', 'testcity',
+            'teststate', '1L2 2G3', 'United Kingdom')
+        self.assertEqual(ag_login_id, obs)
+
+    def test_getAGBarcodeDetails_bad_barcode(self):
+        # test non-existant barcode
+        obs = self.ag_data.getAGBarcodeDetails('99')
+        self.assertEqual(obs, {})
+
+        # test existing barcode but not in AG
+        obs = self.ag_data.getAGBarcodeDetails('000006232')
+        self.assertEqual(obs, {})
 
     def test_getAGBarcodeDetails(self):
-        raise NotImplementedError()
+        # test existing AG barcode
+        obs = self.ag_data.getAGBarcodeDetails('000001047')
+        exp = {
+            'barcode': '000001047',
+            'status': 'Received',
+            'ag_kit_id': 'd8592c74-7e35-2135-e040-8a80115d6401',
+            'name': 'REMOVED',
+            'participant_name': 'REMOVED',
+            'email': 'REMOVED',
+            'site_sampled': 'Stool',
+            'environment_sampled': None,
+            'sample_date': datetime.date(2013, 3, 28),
+            'sample_time': datetime.time(23, 25),
+            'notes': 'REMOVED',
+            'overloaded': None,
+            'withdrawn': None,
+            'other': None,
+            'moldy': None,
+            'refunded': None,
+            'ag_kit_barcode_id': 'd8592c74-7e36-2135-e040-8a80115d6401',
```

Comment:
Those are pulled directly from the database and turned into a dict, so that is whats in the database. Interesting catch.